client/starwhale/core/dataset/type.py

```diff
@@ -702,3 +702,53 @@ def create_by_yaml(cls, fpath: t.Union[str, Path]) -> DatasetConfig:
         c = load_yaml(fpath)
 
         return cls(**c)
+
+
+class JsonDict(SwObject):
+    """
+    JsonDict takes json like dict(https://www.json.org/json-en.html) as init parameter
+    Besides the standard value types, SwObject is an extra value type that is allowed to be passed in
+    """
+
+    def __init__(self, d: dict = {}) -> None:
+        for _k, _v in d.items():
+            if type(_k) != str:
+                raise ValueError("json like dict shouldn't have none-str keys")
+            self.__dict__[_k] = JsonDict.from_data(_v)
+
+    @classmethod
+    def from_data(cls, d: t.Any) -> t.Any:
+        """
+        returns JsonDict or primitive values
+        SwObject is an extra primitive value
+        """
+        if isinstance(d, dict):
+            return cls(d)
+        for _t, _ in _TYPE_DICT.items():
+            if isinstance(d, _t):
+                return d
+        if isinstance(d, SwObject):
+            return d
+        if type(d) == list:
+            return [cls.from_data(_d) for _d in d]
+
+        raise ValueError(
+            "json like dict shouldn't have values who's type is not in [SwObject, list, dict, str, int, bool, None]"
```

Comment:
add `d` value into the exception message?

Reply:
done